engine/server/engine/enclave_manager/enclave_pool.go

```diff
@@ -203,34 +242,58 @@ func (pool *EnclavePool) destroyPreviousIdleEnclaves() error {
 }
 
 // run method is in charge of filling the enclave pool channel with new idle enclaves
-func (pool *EnclavePool) run() {
+func (pool *EnclavePool) run(ctx context.Context) {
 
-	defer close(pool.idleEnclavesChan)
+	defer func() {
+		close(pool.idleEnclavesChan)
```

Comment:
Don't you need to close the `fillChan` at some point also?

Reply:
yes, good catch!